src/OpenTelemetry/Trace/TracerProviderBuilderSdk.cs

```diff
@@ -163,16 +163,18 @@ internal TracerProvider Build()
                     "semver:" + typeof(TInstrumentation).Assembly.GetName().Version,
                     instrumentationFactory));
 
+            this.sources.Add(typeof(TInstrumentation).Assembly.GetName().Name);
+
             return this;
         }
 
         internal readonly struct DiagnosticSourceInstrumentationFactory
         {
             public readonly string Name;
             public readonly string Version;
-            public readonly Func<ActivitySourceAdapter, object> Factory;
+            public readonly Func<object> Factory;
 
-            internal DiagnosticSourceInstrumentationFactory(string name, string version, Func<ActivitySourceAdapter, object> factory)
+            internal DiagnosticSourceInstrumentationFactory(string name, string version, Func<object> factory)
```

Comment:
`DiagnosticSourceInstrumentationFactory` can be totally eliminated now.

Reply:
Removed `DiagnosticSourceInstrumentationFactory`